packages/select/src/Select/components/Overlay/Overlay.tsx

```diff
@@ -1,37 +1,53 @@
 /* eslint-disable jsx-a11y/no-autofocus */
 import React from "react"
+import ReactDOM from "react-dom"
 import { FocusScope } from "@react-aria/focus"
-import { useOverlay, DismissButton } from "@react-aria/overlays"
+import { DismissButton, usePopover } from "@react-aria/overlays"
 import { useSelectContext } from "../../context/SelectContext"
 import styles from "./Overlay.module.scss"
 
 export type OverlayProps = {
   children: React.ReactNode
+  triggerRef: React.RefObject<Element>
 }
 
-export const Overlay = ({ children }: OverlayProps): JSX.Element => {
+export const Overlay = ({
+  children,
+  triggerRef,
+}: OverlayProps): JSX.Element => {
   // Handle events that should cause the menu to close,
   // e.g. blur, clicking outside, or pressing the escape key.
-  const overlayRef = React.useRef<HTMLDivElement>(null)
   const { state } = useSelectContext()
-  const { overlayProps } = useOverlay(
-    { isDismissable: true, isOpen: state.isOpen, onClose: state.close },
-    overlayRef
+
+  const popoverRef = React.useRef<HTMLDivElement>(null)
+  const { popoverProps } = usePopover(
+    {
+      triggerRef,
+      popoverRef,
+      isNonModal: true,
+      placement: "bottom left",
+    },
+    state
   )
 
+  const { width } = triggerRef.current?.getBoundingClientRect() || {}
   // Wrap in <FocusScope> so that focus is restored back to the trigger when the menu is closed
   // and auto focus on the first focusable item after loading. (disable eslint no-autofocus error for it)
   // In addition, add hidden <DismissButton> components at the start and end of the list
   // to allow screen reader users to dismiss the popup easily.
-  return (
-    <div {...overlayProps} ref={overlayRef} className={styles.menuPopup}>
+  const contents = (
+    <div
+      {...popoverProps}
+      ref={popoverRef}
+      style={{ ...popoverProps.style, width }}
+      className={styles.menuPopup}
+    >
       <FocusScope contain autoFocus restoreFocus>
         <DismissButton onDismiss={state.close} />
         {children}
         <DismissButton onDismiss={state.close} />
       </FocusScope>
     </div>
   )
+  return ReactDOM.createPortal(contents, document.body)
```

Comment:
Does it _need_ to be a portal? The rest of our popovers use `usePopper` from `react-popper`. Any reason why we can't use the same pattern?

Reply:
I think primarily the ListBox should be rendered in a Portal as we're trying to avoid the clipping issue, which is probably the preferred way of solving this problem. A popover's solution would be to dynamically position the Listbox in a way that avoids the edge of boundary, but conventionally, a Select's Listbox renders below the input. I have left a note though that currently, the Listbox does flip thanks to React Aria's implementation of a popover, but if Kaizen decides we'd like the prevent flipping, we'd need to portal it anyway. 

Additionally, seeing as the Kaizen Select component has been built almost entirely the same way React Aria suggests, I felt like it made sense to just continue the same pattern - with the exception of using their `<Overlay>` component, which pretty much just abstracts the `React.createPortal` function.